### PR TITLE
Address possible use after free issue in HttpVCTable::remove_entry

### PR DESCRIPTION
**Pull request by shinrich**
Addressing issue #5807

### CODE REVIEW

**Comment by bryancall** [comment]
[approve ci autest]

**Comment by bryancall** [comment]
[approve ci clang-analyzer]

**Comment by bryancall** [comment]
This PR reverts some code that was in this PR #4020.  Isn't there still going to be issues with events happening on stale state machines?

From #4020:
"This PR cleans up any VIOs against the the HttpSM that is being killed. This ends up preventing inactivity timeouts on stale HttpSM's."

**Comment by shinrich** [comment]
Yes, that is why we ran in in production before committing this.  There doesn't seem to be a safe way to reach into the netvc at this point in the code.  To be safe, any terminating netvc would have to be able to reach into this data structure and null out the relevant pointers.  

It seems that we are better off dealing with the stale events by tracking and canceling actions.  Since PR #4020 was committed we also made a number of other fixes to track and cancel actions and to better lock and deliver events to the expected thread.

**Comment by zwoop** [comment]
[approve ci fedora]

**Comment by zwoop** [comment]
This *does* fix the ASAN leak, but as Bryan says, concerning that we might be missing something else that also should have been reverted?

**Comment by zwoop** [comment]
What say ye, should we land this?

File: proxy/http/HttpSM.cc [hunk 1]
```diff
@@ -194,26 +194,10 @@ HttpVCTable::remove_entry(HttpVCTableEntry *e)
...
-  if (e->write_vio != nullptr && e->write_vio->cont == sm) {
-    // Cleanup dangling i/o
-    if (e == sm->get_ua_entry() && sm->get_ua_txn()) {
-      e->write_vio = sm->get_ua_txn()->do_io_write(nullptr, 0, nullptr);
-    } else if (e == sm->get_server_entry() && sm->get_server_session()) {
-      e->write_vio = sm->get_server_session()->do_io_write(nullptr, 0, nullptr);
-    } else {
-      ink_release_assert(false);
-    }
-  }
```
Comment (zwoop) [inline proxy/http/HttpSM.cc]: This is cool, you can do multi-line comments now :).

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Pull request merged**